### PR TITLE
Add python 3.10 to tox and CI

### PR DESCRIPTION
**Pull request by pdecat**
Hi,

I've opened this PR to reproduce some errors I am facing with python 3.10 when running tests.

All of them are like this one:
```
ERROR tests/test_xiaomi.py::test_xiaomi_battery[2985-118] - TypeError: Can't instantiate abstract class MockApp with abstract methods reset_network_info, send_packet
```

Note: python 3.10 is supported by Home Assistant since 2022/07, see https://www.home-assistant.io/blog/2022/07/06/release-20227/#improved-stability-and-performance-and-python-310

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
 
 env:
   CACHE_VERSION: 1
-  DEFAULT_PYTHON: 3.7
+  DEFAULT_PYTHON: 3.9
```
Comment (pdecat) [inline .github/workflows/ci.yml:13]: Should we target 3.10 now?
```suggestion
  DEFAULT_PYTHON: "3.10"
```

Comment (pdecat) [reply .github/workflows/ci.yml:13]: Note: zigpy is going to drop 3.7 too, but still target 3.8.

https://github.com/zigpy/zigpy-zigate/pull/131/commits/5f1fdc1aa8527fe9bb43b4210912125b78695e93

Comment (pdecat) [reply .github/workflows/ci.yml:13]: Should we pin versions exactly like zigpy?
```suggestion
  DEFAULT_PYTHON: 3.9.14
```
https://github.com/zigpy/zigpy/pull/1050/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR10

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, "3.10"]
```
Comment (pdecat) [inline .github/workflows/ci.yml:24]: Should we pin versions exactly like zigpy?
```suggestion
        python-version: ['3.8.14', '3.9.14', '3.10.7']
```
https://github.com/zigpy/zigpy/pull/1050/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR21

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -280,7 +280,7 @@ jobs:
     needs: prepare-base
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.8, 3.9, "3.10"]
```
Comment (pdecat) [inline .github/workflows/ci.yml:283]: Should we pin versions exactly like zigpy?
```suggestion
        python-version: ['3.8.14', '3.9.14', '3.10.7']
```
https://github.com/zigpy/zigpy/pull/1050/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR21

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dmulcahey) [reply .github/workflows/ci.yml:13]: Yes

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dmulcahey) [reply .github/workflows/ci.yml:24]: Yes

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dmulcahey) [reply .github/workflows/ci.yml:283]: Yes

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dmulcahey) [reply .github/workflows/ci.yml:13]: We should match Zigpy’s targets

File: .github/workflows/ci.yml (hunk 3 above)
Comment (pdecat) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pdecat) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pdecat) [reply .github/workflows/ci.yml]: Done.

**Pull request merged**